app/scripts/lib/multichain-api/scope/assert.ts

```diff
@@ -0,0 +1,103 @@
+import MetaMaskOpenRPCDocument from '@metamask/api-specs';
+import { NetworkClientId } from '@metamask/network-controller';
+import { Hex } from '@metamask/utils';
+import { InternalAccount } from '@metamask/keyring-api';
+import { EthereumRpcError } from 'eth-rpc-errors';
+import {
+  isSupportedAccount,
+  isSupportedNotification,
+  isSupportedScopeString,
+} from './supported';
+import { ScopeObject, ScopesObject } from './scope';
+
+const validRpcMethods = MetaMaskOpenRPCDocument.methods.map(({ name }) => name);
+
+export const assertScopeSupported = (
+  scopeString: string,
+  scopeObject: ScopeObject,
+  {
+    findNetworkClientIdByChainId,
+    getInternalAccounts,
+  }: {
+    findNetworkClientIdByChainId: (chainId: Hex) => NetworkClientId;
+    getInternalAccounts: () => InternalAccount[];
+  },
+) => {
+  const { methods, notifications, accounts } = scopeObject;
+  if (!isSupportedScopeString(scopeString, findNetworkClientIdByChainId)) {
+    throw new EthereumRpcError(5100, 'Requested chains are not supported');
+  }
+
+  // Needs to be split by namespace?
+  const allMethodsSupported = methods.every((method) =>
+    validRpcMethods.includes(method),
+  );
+  if (!allMethodsSupported) {
+    // not sure which one of these to use
+    // When provider evaluates requested methods to not be supported
+    //   code = 5101
+    //   message = "Requested methods are not supported"
+    // When provider does not recognize one or more requested method(s)
+    //   code = 5201
+    //   message = "Unknown method(s) requested"
+
+    throw new EthereumRpcError(5101, 'Requested methods are not supported');
+  }
+
+  if (
+    notifications &&
+    !notifications.every((notification) =>
+      isSupportedNotification(notification),
+    )
+  ) {
+    // not sure which one of these to use
+    // When provider evaluates requested notifications to not be supported
+    //   code = 5102
+    //   message = "Requested notifications are not supported"
+    // When provider does not recognize one or more requested notification(s)
+    //   code = 5202
+    //   message = "Unknown notification(s) requested"
+    throw new EthereumRpcError(
+      5102,
+      'Requested notifications are not supported',
+    );
+  }
+
+  if (accounts) {
+    const accountsSupported = accounts.every((account) =>
+      isSupportedAccount(account, getInternalAccounts),
+    );
+
+    if (!accountsSupported) {
+      // TODO: There is no error code or message specified in the CAIP-25 spec for when accounts are not supported
+      // The below is made up
+      throw new EthereumRpcError(5103, 'Requested accounts are not supported');
```

Comment:
weren't we going to get rid of this?

Reply:
i commented out the validation part so it doesnt get called. also a little bit weird to throw a jsonrpc specific error from this helper since it could get used in validation for the caveats